inventory.py

```diff
@@ -105,47 +90,47 @@ def handle_request_if_failed(response):
 def get_course_publish_date(df_row: str, publish_date, next_page_url = None)->str:
     logger.debug(df_row)
     if df_row['course_workflow_state'] != 'available':
-        return 'null'
-
-    course_id= df_row['course_id']
+        return None
 
-    if next_page_url is not None:
-        url = next_page_url
-    else:
-        url = f"{API_SCOPE_PREFIX}/audit/course/courses/{course_id}?per_page={PER_PAGE_COUNT}"
+    course_id = df_row['course_id']
+    url = f"{API_SCOPE_PREFIX}/audit/course/courses/{course_id}?per_page={PER_PAGE_COUNT}"
 
     try:
-        response = make_audit_logs_api_request(url)
+        if next_page_url is not None:
+            response = make_audit_logs_api_request(url, next_page_url)
+        else:
+            response = make_audit_logs_api_request(url)
 
-    except (Exception) as e:
+    except Exception as e:
         logger.exception('getting published date for a course has erroneous response ' + e.message)
-        return publish_date
+        return None
 
     if not handle_request_if_failed(response):
-        return publish_date
+        return None
 
     audit_events = json.loads(response.text.encode('utf8'))
     if not audit_events:
-        return publish_date
+        return None
 
     events = audit_events['events']
 
     for event in events:
         if event['event_type'] == 'published':
-            publish_date = event['created_at']
-            logger.info(f"Date for Workflow type :{df_row['course_workflow_state']} for course {course_id}")
+            publish_date.append(event['created_at'])
+            logger.info(f"Date for Workflow type {df_row['course_workflow_state']} for course {course_id}")
             break
-    if publish_date is None:
-        next_page_url = get_next_page_url(response)
+    if len(publish_date) == 0:
+        next_page_url = API_UTIL.get_next_page(response)
```

Comment:
Changed the next page logic based on test case in the API utils https://github.com/tl-its-umich-edu/api-utils-python/blob/25aec8f0228cc426d463d5958428bbbc6ddd6d33/tests/test.py#L73.

I don't think the previous implementation of mine worked. But Now It will :)